internals/src/Tizen.Applications.ThemeManager/Tizen.Applications.ThemeManager/Theme.cs

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Tizen.Applications.ThemeManager
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class Theme : IDisposable
+    {
+        private bool _disposed = false;
+        private IntPtr _handle;
+        private string _id = String.Empty;
+        private string _version = String.Empty;
+        private string _toolVersion = String.Empty;
+        private string _title = String.Empty;
+        private string _resolution = String.Empty;
+        private string _preview = String.Empty;
+        private string _description = String.Empty;
+
+        /// <summary>
+        /// A copy constructor of Theme.
+        /// </summary>
+        /// <param name="theme">Theme class.</param>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed because of system error.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of out of memory.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public Theme(Theme theme)
+        {
+            if (theme == null || theme._handle == null)
+            {
+                Interop.ThemeManager.ThemeManagerErrorFactory.GetException(Interop.ThemeManager.ErrorCode.InvalidParameter, "Invalid parameter");
+            }
+
+            var err = Interop.ThemeManager.ErrorCode.None;
+            err = Interop.ThemeManager.ThemeClone(theme._handle, out _handle);
+            if (err != Interop.ThemeManager.ErrorCode.None)
+            {
+                Interop.ThemeManager.ThemeManagerErrorFactory.GetException(err, "Failed to clone handle");
+            }
+        }
+
+        internal Theme(IntPtr handle)
```

Comment:
The handle from the callback will not be valid after the callback is called.
In this case, it should not be destroyed. How about using owner flag or cloning it?

Reply:
You should cloning a handle. because usually C# EventArgs can be used out of the event handler scope

Reply:
Done